### DISCUSSION TITLE
Component code (script setup) executed on both server and client on page refresh

### DISCUSSION THREAD

**Discussion by danielrjames** [nuxt/nuxt#17712] [Questions] [answered]
I have an index page script setup component with a console.log. Whenever I refresh the page, this gets executed on both server and client.

If I wrap this console.log in useAsyncData, it will only get executed on the server (as expected). Navigating to this page results in this only being executed on the client (also expected).

I did come across a [similar thread](https://github.com/nuxt/framework/discussions/2294) and the mentioned solution was to provide a .favicon.ico, so I did add that and the issue still persists.

Was wondering if this is a bug or is somehow intentional?

```
<script setup>
  const instance = process.client ? 'client' : 'server';

  useAsyncData(useRoute().name, () => {
    console.log('asyncData instance', instance); // logs on the server on page refresh
  });

  console.log('outside asyncData', instance); // logs on both server and client on page refresh
</script>
```

**Comment by danielroe** [comment, marked as answer]
This is correct behaviour. The `setup` lifecycle hook runs on both server and client.

**Reply by danielrjames** [reply]
That's good to know it's not a bug.

What's the benefit of running on both? Do the docs cover this? I'd like to read more on it.
